### DISCUSSION TITLE
Poly project sync for 3rd party dependencies?

### DISCUSSION THREAD

**Discussion by saifmode** [DavidVujic/python-polylith#286] [Q&A] [answered]
Found out about polylith today, it's the perfect tool for my use case! I'm enjoying it a lot so far.

However, one snag I've run into is syncing my 'project' with the third party libraries used by its bricks. The `poly sync` command automatically adds bricks to the project, which is great. But it doesn't add third party libraries. Are we expected to manually add these to the project's pyproject.toml by hand?

**Comment by DavidVujic** [comment, marked as answer]
Hi @saifmode, I'm happy that you have found Polylith and that it is a useful tool!

Yes, the third-party libraries are added manually. One reason is that libraries aren't that often changed in comparison to your source code, that is constantly developed and refactored. I also think libraries are something you as a developer want more control over, and probably prefer to use the dependency management tool in the project (like Poetry, Hatch or uv) - that also handles the lock-files.

The Polylith tool has the `poly check` command and the `poly libs` command to notify about any missing libraries. The `poly check` command is also useful in a CI, where it will return with an error exit code if there is any missing library.

**Comment by saifmode** [comment]
Cool, that makes sense.

I wonder if a nice automation would be to optionally add these third party libs to a project? So doing:

`poly sync --directory projects/<my project> --auto-lib`

Would do sync the bricks etc, but also run poly check for missing 3rd party libs, and pull them from the root project toml if they exist. Might be a bad idea, but interested to hear your thoughts.

**Reply by DavidVujic** [reply]
I think the tool would need to do something with the lock file(s) too, or run the chosen dependency management tool in a subprocess (poetry, hatch, uv ...) to have everything correctly updated, right?

I would recommend to develop a custom script that does that, by combining the poly tool and poetry/hatch/uv.
